### PR TITLE
Improve NEO event handling

### PR DESCRIPTION
**Pull request by alejoe91**
+ extend open ephys events to handle block_index and test with different datasets

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Ok for me.
I like it.
But, if we go for this, I would change the name `get_event_times()` to get_events()`.

Or maybe have both:
 *  to get_events()` : return complex dtype with 'time' include
 *  `get_event_times()` : return only time as float64 Easy to write on top of `get_events()`

**Comment by alejoe91** [comment]
> Ok for me. I like it. But, if we go for this, I would change the name `get_event_times()` to get_events()`.
> 
> Or maybe have both:
> 
> * to get_events()` : return complex dtype with 'time' include
> * `get_event_times()` : return only time as float64 Easy to write on top of `get_events()`

I agree! Will make a follow-up PR

**Comment by alejoe91** [comment]
@samuelgarcia ready to review. I restructured a bit and cleaned up the base as well. Let me know what you think

File: spikeinterface/extractors/neoextractors/neobaseextractor.py [hunk 1]
```diff
@@ -317,16 +320,17 @@ def __init__(self, block_index=None, **neo_kwargs):
 
         event_channels = self.neo_reader.header['event_channels']
 
-        channel_ids = event_channels['id']
+        channel_ids = event_channels['name']
```
Comment (alejoe91) [inline spikeinterface/extractors/neoextractors/neobaseextractor.py:323]: check unique

Comment (alejoe91) [inline spikeinterface/extractors/neoextractors/neobaseextractor.py:323]:
```suggestion
        channel_ids = event_channels['id']
```

**Pull request merged**